src/scry/client.cr

```diff
@@ -0,0 +1,56 @@
+require "./protocol/initialize"
+require "./protocol/response_message"
+require "./protocol/notification_message"
+
+module Scry
+  class Client
+    alias ClientMessage = Initialize | ResponseMessage | NotificationMessage | Nil
+
+    def self.instance(io : IO = STDOUT)
```

Comment:
I think this is a bit tricky if you want to use your own IO (i.e: in specs), you _need_ to make sure that the first call to `Client.instance` has an IO, and you _can't change_ the IO later once it's set.

I think it would be better to keep the `Client.instance` as a simple getter, and change the architecture: maybe `Client` doesn't even need to be a singleton?